### PR TITLE
Test cases for plio app

### LINKED ISSUE #193: More tests for plio app

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #193 

## Summary
- tests for several endpoints in the `plio`, `item`, `question` and `image` tables.

## Test Plan
- [x] Wrote tests
- [x] Tested locally
- [ ] Tested on staging
- [ ] Tested on production
- [x] ~If changes in DB, update [DB schema](https://docs.google.com/presentation/d/15c45HbHphh4w3guaFV3BgEA5nMKd-4J35JdUKdrpPzo/edit#slide=id.gc8c9ec0b5e_0_0)~

### CODE REVIEW

File: plio/tests.py [hunk 1]
```diff
@@ -268,3 +652,19 @@ def test_each_image_has_unique_name(self):
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        image = Image.objects.filter(id=self.image).first()
+        new_image = Image.objects.filter(id=response.data["id"]).first()
+
+        self.assertNotEqual(self.image, response.data["id"])
+        self.assertEqual(image.alt_text, response.data["alt_text"])
+        self.assertEqual(image.url.file.size, new_image.url.file.size)
+
+    # before merge, update Vaibhav's changes for org and new user - use
```
Comment (dalmia) [inline plio/tests.py:669]: this has been kept as a reminder based on whether Vaibhav's or this PR gets merged first. Will remove it before merging.

File: plio/tests.py [hunk 2]
```diff
@@ -139,18 +310,244 @@ class ItemTestCase(BaseTestCase):
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # the number of responses should still be 1 - the item created above
+        # should not be included
+        self.assertEqual(len(response.data), 1)
+
+    def test_user_list_own_items(self):
+        """A user should only be able to list their own items"""
+        # create plio from a different user
+        new_plio = Plio.objects.create(
```
Comment (deepansh96) [inline plio/tests.py:356]: If writing this test, the test `test_user_can_list_items` is redundant right?

File: plio/tests.py [hunk 3]
```diff
@@ -139,18 +310,244 @@ class ItemTestCase(BaseTestCase):
...
+        # create item from a different user
+        Item.objects.create(type="question", plio=new_plio, time=1)
+
+        # get items
+        response = self.client.get(reverse("items-list"))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        # the count should remain 1 as the new item was created with different user
+        self.assertEqual(len(response.data), 1)
+
+    def test_user_can_list_own_items_org(self):
```
Comment (deepansh96) [inline plio/tests.py:369]:
```suggestion
    def test_org_user_can_list_own_items(self):
```
I think this is more clear

File: plio/tests.py [hunk 4]
```diff
@@ -139,18 +310,244 @@ class ItemTestCase(BaseTestCase):
...
+
+        item_org = Item.objects.create(type="question", plio=plio_org, time=1)
+
+        # set organization in request
+        self.client.credentials(
+            HTTP_ORGANIZATION=self.organization.shortcode,
+            HTTP_AUTHORIZATION="Bearer " + self.access_token.token,
+        )
+
+        # get plios
```
Comment (deepansh96) [inline plio/tests.py:396]:
```suggestion
        # get items
```

File: plio/tests.py [hunk 5]
```diff
@@ -139,18 +310,244 @@ class ItemTestCase(BaseTestCase):
...
+        response = self.client.get(reverse("items-list"))
+
+        # the plio created above should be listed
```
Comment (deepansh96) [inline plio/tests.py:399]:
```suggestion
        # the item created above should be listed
```

File: plio/tests.py [hunk 6]
```diff
@@ -139,18 +310,244 @@ class ItemTestCase(BaseTestCase):
...
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["id"], item_org.id)
+
+        # set db connection back to public (default) schema
+        connection.set_schema_to_public()
+
+    def test_user_can_list_other_items_org(self):
```
Comment (deepansh96) [inline plio/tests.py:406]:
```suggestion
    def test_org_user_can_list_other_user_items(self):
```

File: plio/tests.py [hunk 7]
```diff
@@ -139,18 +310,244 @@ class ItemTestCase(BaseTestCase):
...
+        response = self.client.post(
+            f"/api/v1/questions/{self.question.id}/duplicate/",
+            {"itemId": self.item.id + 100},
+        )
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            json.loads(response.content)["detail"], "Specified item not found"
+        )
+
+    def test_duplicate_own_question(self):
```
Comment (deepansh96) [inline plio/tests.py:513]: `test_cannot_duplicate_other_user_question`
This test case is not there right?

**Review by deepansh96 — CHANGES_REQUESTED** [review]

**Pull request merged**